pkg/tuple/tuple.go

```diff
@@ -10,6 +10,18 @@ import (
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
+type TupleWithCondition interface {
+	TupleWithoutCondition
+	GetCondition() *openfgav1.RelationshipCondition
+}
+
+type TupleWithoutCondition interface {
+	GetUser() string
+	GetObject() string
+	GetRelation() string
+	String() string
```

Comment:
I don't think you should put `String` here.  I haven't tested it but i think`TupleWithCondition.String()` will delegate to `TupleWithoutCondition.String()`

Reply:
That's intended. If you're composing tuples a TupleWithCondition is a TupleWithoutCondition + the GetCondition bit. So the TupleWithoutCondition is the base triplet and we have a string method for it. See it's usage. It's only used where we print `(object, relation, user)`, and its default implementation comes from the protobuf generated `String` method fo the `TupleKeyWithoutCondition`.